### PR TITLE
[wpmlcore-7279] Search widget fix with language as parameter

### PR DESCRIPTION
**Pull request by GeorgeBotsev**
[wpmlcore-7279] Add new class that is especially for Elementor search widget to handle different language search via hidden language parameter.
Add new class to be loaded on frontend actions

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-7279

### CODE REVIEW

File: src/Hooks/Frontend.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace WPML\PB\Elementor\Hooks;
+
+class Frontend implements \IWPML_Frontend_Action, \IWPML_DIC_Action {
+
+    public function add_hooks() {
+        add_action( 'elementor_pro/search_form/after_input', [ $this, 'elementor_add_language_form_field' ] );
+    }
+
+    public function elementor_add_language_form_field() {
```
Comment (strategio) [inline src/Hooks/Frontend.php]: @GeorgeBotsev, for new code (and when it's not forced by an interface), we want to use camelCase. Could you please rename if `addLanguageFormField`?

Note: We don't need to mention `elementor` in the method name, this is implicit.

File: tests/phpunit/tests/Hooks/TestFrontend.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+	/**
+	 * @test
+	 */
+	public function TestAddHooks() {
+		$frontend = new Frontend();
+		\WP_Mock::expectActionAdded( 'elementor_pro/search_form/after_input', [ $frontend, 'elementor_add_language_form_field' ] );
+		$frontend->add_hooks();
+    }
+}
```
Comment (strategio) [inline tests/phpunit/tests/Hooks/TestFrontend.php:29]: @GeorgeBotsev, if you want to cover the whole code (and we want this :), you also need a test on the method `addLanguageFormField`:

```php
/**
 * @test 
 */
public function itShouldAddLanguageFormField() {
     $subject = new Frontend();
    
    \WP_Mock::expectAction( 'wpml_add_language_form_field' );

    $subject->add_hooks();
}
```

File: tests/phpunit/tests/test-wpml-elementor-integration-factory.php [hunk 3]
```diff
@@ -28,7 +28,8 @@ public function it_creates() {
 			                     'WPML_Elementor_Media_Hooks_Factory',
 			                     'WPML_Elementor_WooCommerce_Hooks_Factory',
 			                     \WPML\PB\Elementor\LanguageSwitcher\LanguageSwitcher::class,
-			                     \WPML\PB\Elementor\Hooks\DynamicElements::class,
+								 \WPML\PB\Elementor\Hooks\DynamicElements::class,
+								 \WPML\PB\Elementor\Hooks\Frontend::class,
```
Comment (strategio) [inline tests/phpunit/tests/test-wpml-elementor-integration-factory.php]: There's something wrong with the indentation here @GeorgeBotsev.

**Review by strategio — CHANGES_REQUESTED** [review]

File: tests/phpunit/tests/Hooks/TestFrontend.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+		$subject->add_hooks();
+	}
+
+	/**
+	 * @test
+	 */
+	public function itShouldAddLanguageFormField() {
+		$subject = new Frontend();
+	   \WP_Mock::expectAction( 'wpml_add_language_form_field' );
+	   $subject->add_hooks();
```
Comment (strategio) [inline tests/phpunit/tests/Hooks/TestFrontend.php]: @GeorgeBotsev, this is not correct (and tests are failing). You need to call the `addLanguageFormField` method:

```
$subject->addLanguageFormField();
```

Sorry, the example in my previous comment was wrong...

Comment (strategio) [reply tests/phpunit/tests/Hooks/TestFrontend.php]: And please check the indentation.

**Review by strategio — DISMISSED** [review]
+1 @GeorgeBotsev

File: src/Hooks/Frontend.php [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace WPML\PB\Elementor\Hooks;
+
+class Frontend implements \IWPML_Frontend_Action, \IWPML_DIC_Action {
```
Comment (dgwatkins) [inline src/Hooks/Frontend.php]: @GeorgeBotsev I don't think we need `IWPML_DIC_Action` here because there is no constructor to inject dependencies to.

Comment (GeorgeBotsev) [reply src/Hooks/Frontend.php]: I was not sure about that.
I tested  now and it seems that you are right, so removing this now.

**Review by dgwatkins — DISMISSED** [review]
Nice @GeorgeBotsev

**Review by strategio — DISMISSED** [review]
Well done @GeorgeBotsev +1

**Comment by dimitrismitsis** [comment]
@GeorgeBotsev This one could use some rebasing and conflict-resolving too!

**Pull request merged**